src/main/java/org/entando/kubernetes/service/digitalexchange/component/EntandoBundleServiceImpl.java

```diff
@@ -86,13 +89,47 @@ public EntandoBundleServiceImpl(K8SServiceClient k8SServiceClient,
     private List<EntandoBundle> listAllBundles() {
         List<EntandoBundle> allComponents = new ArrayList<>();
         List<EntandoBundleEntity> installedBundles = installedComponentRepo.findAll();
+
         List<EntandoBundle> availableBundles = listBundlesFromEcr();
         List<EntandoBundle> installedButNotAvailableOnEcr = filterInstalledButNotAvailableOnEcr(availableBundles,
                 installedBundles);
 
         allComponents.addAll(availableBundles);
         allComponents.addAll(installedButNotAvailableOnEcr);
-        return allComponents;
+
+        return populateCustomInstallation(allComponents, installedBundles);
+    }
+
+    /**
+     * for each installed bundle, populate the field custom_installation in the allBundles list.
+     * @param allBundles the list representing all the available EntandoBundle
+     * @param installedBundles the list of installed EntandoBundleEntity
+     * @return a new instance of a list of all bundles with customInstallation field populated
+     */
+    private List<EntandoBundle> populateCustomInstallation(List<EntandoBundle> allBundles, List<EntandoBundleEntity> installedBundles) {
```

Comment:
@firegloves I think populates essentially means "filling".
What about a simple `updateCustomInstallationFlag`?

Reply:
it sounds good